pkg/snapstore/abs_snapstore.go

```diff
@@ -56,32 +58,59 @@ type absCredentials struct {
 	StorageAccount string `json:"storageAccount"`
 }
 
-// NewABSSnapStore create new ABSSnapStore from shared configuration with specified bucket
+// NewABSSnapStore creates a new ABSSnapStore using a shared configuration and a specified bucket
 func NewABSSnapStore(config *brtypes.SnapstoreConfig) (*ABSSnapStore, error) {
 	storageAccount, storageKey, err := getCredentials(getEnvPrefixString(config.IsSource))
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to get credentials: %v", err)
 	}
 
 	credentials, err := azblob.NewSharedKeyCredential(storageAccount, storageKey)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create shared key credentials: %v", err)
 	}
 
-	p := azblob.NewPipeline(credentials, azblob.PipelineOptions{
+	pipeline := azblob.NewPipeline(credentials, azblob.PipelineOptions{
 		Retry: azblob.RetryOptions{
 			TryTimeout: downloadTimeout,
 		}})
-	u, err := url.Parse(fmt.Sprintf("https://%s.%s", storageAccount, brtypes.AzureBlobStorageHostName))
+
+	blobURL, err := ConstructBlobServiceURL(credentials)
 	if err != nil {
-		return nil, fmt.Errorf("failed to parse service url: %v", err)
+		return nil, err
 	}
-	serviceURL := azblob.NewServiceURL(*u, p)
+
+	serviceURL := azblob.NewServiceURL(*blobURL, pipeline)
 	containerURL := serviceURL.NewContainerURL(config.Container)
 
 	return GetABSSnapstoreFromClient(config.Container, config.Prefix, config.TempDir, config.MaxParallelChunkUploads, config.MinChunkSize, &containerURL)
 }
 
+// ConstructBlobServiceURL constructs the Blob Service URL based on whether the Azurite Emulator is enabled or not
+func ConstructBlobServiceURL(credentials *azblob.SharedKeyCredential) (*url.URL, error) {
+	defaultURL, err := url.Parse(fmt.Sprintf("https://%s.%s", credentials.AccountName(), brtypes.AzureBlobStorageHostName))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse default service URL: %v", err)
+	}
+
+	if emulatorEnabled, ok := os.LookupEnv(envEmulatorEnabled); ok {
+		isEmulator, err := strconv.ParseBool(emulatorEnabled)
+		if err != nil {
+			return nil, fmt.Errorf("invalid value for %s: %s, error: %v", envEmulatorEnabled, emulatorEnabled, err)
+		}
+
+		if isEmulator {
+			if endpoint, ok := os.LookupEnv(azuriteEndpoint); ok {
+				// application protocol is set by the user of Azurite, not by backup-restore
+				return url.Parse(fmt.Sprintf("%s/%s", endpoint, credentials.AccountName()))
+			}
+			return nil, fmt.Errorf("%s environment variable not set while %s is true", azuriteEndpoint, envEmulatorEnabled)
+		}
+	}
+
+	return defaultURL, nil
+}
+
```

Comment:
```suggestion
// ConstructBlobServiceURL constructs the Blob Service URL based on the activation status of the Azurite Emulator.
// It checks the environment variables for emulator configuration and constructs the URL accordingly.
// The function expects two environment variables:
// - EMULATOR_ENABLED: Indicates whether the Azurite Emulator is enabled (expects "true" or "false").
// - AZURE_STORAGE_API_ENDPOINT: Specifies the Azurite Emulator endpoint when the emulator is enabled.
func ConstructBlobServiceURL(credentials *azblob.SharedKeyCredential) (*url.URL, error) {
	defaultURL, err := url.Parse(fmt.Sprintf("https://%s.%s", credentials.AccountName(), brtypes.AzureBlobStorageHostName)) // Assuming direct string for clarity.
	if err != nil {
		return nil, fmt.Errorf("failed to parse default service URL: %w", err)
	}

	// Check if the Azurite Emulator is enabled via environment variables.
	emulatorEnabled, ok := os.LookupEnv(envEmulatorEnabled)
	if !ok {
		// If the EMULATOR_ENABLED environment variable is not set, return the default URL.
		return defaultURL, nil
	}

	isEmulator, err := strconv.ParseBool(emulatorEnabled)
	if err != nil {
		return nil, fmt.Errorf("invalid value for %s: %s, error: %w", envEmulatorEnabled, emulatorEnabled, err)
	}

	// If the emulator is not enabled, return the default service URL.
	if !isEmulator {
		return defaultURL, nil
	}

	// If the emulator is enabled, construct the URL using the emulator endpoint from the environment variables.
	endpoint, ok := os.LookupEnv(azuriteEndpoint)
	if !ok {
		return nil, fmt.Errorf("%s environment variable not set while %s is true", azuriteEndpoint, envEmulatorEnabled)
	}

	// Application protocol (http or https) is determined by the user of the Azurite, not by this function.
	return url.Parse(fmt.Sprintf("%s/%s", endpoint, credentials.AccountName()))
}

```

Reply:
Addressed in [6fca638](https://github.com/gardener/etcd-backup-restore/pull/699/commits/6fca638920bac61b1739494207b0f965596836c7).